docs/posts/ibisml/index.qmd

```diff
@@ -0,0 +1,897 @@
+---
+title: "Using IbisML for a Kagge competition: credit risk model stability"
+author: "Jiting Xu"
+date: "2024-07-25"
+categories:
+    - blog
+    - machine learning
+    - feature engineering
+execute:
+    freeze: auto
+---
+
+## Introduction
+In this post, we'll demonstrate how to use Ibis and IbisML end-to-end for a Kaggle competition -
+[Credit risk model stability](https://www.kaggle.com/competitions/home-credit-credit-risk-model-stability).
+
+1. Load data and perform feature engineering with Ibis on a duckdb backend
+2. Conduct last-mile ML data preprocessing with IbisML
+3. Train two models using different frameworks:
+    * An XGBoost model within a scikit-learn pipeline.
+    * A neural network with PyTorch and PyTorch Lightning
+
+The aim of this competition is to predict which clients are more likely to default on their
+loans by using both internal and external data sources.
+
+To get started with Ibis and IbisML, please refer to the websites:
+
+* [Ibis](https://ibis-project.org/): An open-source dataframe library that works with any data system.
+* [IbisML](https://ibis-project.github.io/ibis-ml/): A library for building scalable ML pipelines.
+
+
+## Prerequisites
+To run this example, you'll need to download the data from Kaggle website with a Kaggle user account and install Ibis, IbisML, and the necessary modeling library.
+
+### Download data
+You need a Kaggle account to download the data. If you do not have one,
+feel free to register one.
+
+1. Option 1: Manual Download
+     * Log into your Kaggle account and download all data from this
+     [link](https://www.kaggle.com/competitions/home-credit-credit-risk-model-stability/data),
+     unzip the files, and save them to your local disk.
+2. Option 2: Kaggle API
+    * Go to your Kaggle account settings: [Kaggle Account Settings](https://www.kaggle.com/account).
+    * Under the "API" section, click on "Create New API Token". This will download the `kaggle.json`
+    file to your computer.
+    * Place the `kaggle.json` file in the correct directory, normally it is under your home directory
+    `~/.kaggle`:
+
+        ```bash
+        mkdir ~/.kaggle
+        mv ~/Downloads/kaggle.json ~/.kaggle
+        ```
+    * Install Kaggle CLI and download the data:
+
+        ```bash
+        pip install kaggle
+        kaggle competitions download -c home-credit-credit-risk-model-stability
+        unzip home-credit-credit-risk-model-stability.zip
+        ```
+
+### Install libraries
+To use Ibis and IbisML with the DuckDB backend, you'll need to install the necessary packages:
+```{.bash}
+pip install 'ibis-framework[duckdb]' ibis-ml
+```
+
+When building models, ensure you have the required libraries installed:
+```{.bash}
+# PyTorch
+pip install torch torchvision pytorch-lightning
+
+# XGBoost and sklearn
+pip install sklearn xgboost
```

Comment:
Nit: Personal pet peeve is when you have the `pip install` broken into multiple steps; you could consider giving two options:

* `pip install 'ibis-framework[duckdb]' ibis-ml torch torchvision pytorch-lightning`
* `pip install 'ibis-framework[duckdb]' ibis-ml sklearn xgboost`

Nit: I prefer `xgboost[scikit-learn]` to `sklearn xgboost`, because that way you better guarantee compatibility.

Also, bit surprised `torchvision` is required here?